File: Backend/src/main/java/com/tripnest/analytics/service/AnalyticsService.java
package com.tripnest.analytics.service;

import com.tripnest.analytics.dto.DashboardStatsResponse;

public interface AnalyticsService {
    DashboardStatsResponse getDashboardStats(String username);
    com.tripnest.analytics.dto.AnalyticsReportDto getTripAnalytics(Long tripId, String username);
    com.tripnest.analytics.dto.AnalyticsReportDto getUserGlobalAnalytics(String username);
}

File: frontend/src/services/analyticsService.js
import { apiClient } from './apiClient';

export const analyticsService = {
  getDashboardStats: async () => {
    try {
      return await apiClient.get('/analytics/dashboard/stats');
    } catch (error) {
      console.error('Failed to get dashboard stats:', error);
      return null;
    }
  },

  getTripAnalytics: async (tripId) => {
    try {
      return await apiClient.get(`/analytics/trips/${tripId}`);
    } catch (error) {
      console.error(`Failed to get trip analytics for ${tripId}:`, error);
      return null;
    }
  },

  getUserGlobalAnalytics: async () => {
    try {
      return await apiClient.get('/analytics/reports');
    } catch (error) {
      console.error('Failed to get global analytics:', error);
      return null;
    }
  },

  getAdminAnalytics: async () => {
    try {
      return await apiClient.get('/analytics/admin');
    } catch (error) {
      console.error('Failed to get admin analytics:', error);
      return null;
    }
  }
};
